src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -283,6 +294,62 @@ public void InitializeService(ServiceHost serviceHost, SqlToolsContext context)
 
         #region Request Handlers
 
+        /// <summary>
+        /// Completion extension load request callback
+        /// </summary>
+        /// <param name="param"></param>
+        /// <param name="requestContext"></param>
+        /// <returns></returns>
+        internal async Task HandleCompletionExtLoadRequest(CompletionExtensionParams param, RequestContext<bool> requestContext)
+        {
+            int extCount = 0;
+            try
+            {
+                //register the new assembly
+                var serviceProvider = (ExtensionServiceProvider)ServiceHostInstance.ServiceProvider;
+                var assemblies = new Assembly[] { AssemblyLoadContext.Default.LoadFromAssemblyPath(param.Assembly) };
+                serviceProvider.AddAssembliesToConfiguration(assemblies);
+                foreach (var provider in serviceProvider.GetServices<ICompletionExtensionProvider>())
```

Comment:
>provider [](start = 29, length = 8)

Shouldn't this be var service instead? You're getting the services from the provider - not other providers.

Reply:
Since we are using MEF for service discovery, I think I should drop ICompletionExtensionProvider, what do you think?

Reply:
I think I can drop ICompletionExtensionProvider interface, what do you think?

Reply:
Oh yeah good point - definitely doesn't seem useful anymore. Sounds good to me!

---
In reply to: [304700466](https://github.com/microsoft/sqltoolsservice/pull/833#discussion_r304700466) [](ancestors = 304700466)